sdk/static_files.go

```diff
@@ -28,13 +30,19 @@ func (staticfile *StaticFiles) GetName() string {
 
 //GetPath returns the path of the artifact
 func (staticfile *StaticFiles) GetPath() string {
-	container := base64.RawURLEncoding.EncodeToString([]byte(fmt.Sprintf("%d-%s", staticfile.NodeJobRunID, url.PathEscape(staticfile.Name))))
+	var container string
+	if staticfile.StaticKey != "" {
+		container = base64.RawURLEncoding.EncodeToString([]byte(fmt.Sprintf("%d-%d-%s", staticfile.WorkflowID, staticfile.StaticKey, url.PathEscape(staticfile.Name))))
+	} else {
+		container = base64.RawURLEncoding.EncodeToString([]byte(fmt.Sprintf("%d-%s", staticfile.NodeJobRunID, url.PathEscape(staticfile.Name))))
+	}
 	return container
```

Comment:
return base64.RawURLEncoding.EncodeToString([]byte(container)) ?
to clean code before